android/compose/src/main/kotlin/com/boswelja/ephemeris/compose/EphemerisCalendar.kt

```diff
@@ -9,36 +9,21 @@ import androidx.compose.foundation.layout.Row
 import androidx.compose.foundation.layout.fillMaxWidth
 import androidx.compose.runtime.Composable
 import androidx.compose.runtime.remember
-import androidx.compose.runtime.rememberCoroutineScope
 import androidx.compose.ui.Modifier
 import com.boswelja.ephemeris.core.model.CalendarDay
 import com.boswelja.ephemeris.core.model.CalendarPage
-import com.boswelja.ephemeris.core.ui.CalendarPageLoader
 
 @OptIn(ExperimentalAnimationApi::class)
 @Composable
 public fun EphemerisCalendar(
-    calendarState: CalendarState,
+    calendarState: CalendarStateImpl,
```

Comment:
Passing the implementation explicitly instead of the interface?

Reply:
Yep, a couple of reasons
1. Interface doesn't allow for anything other than public, and we don't want to expose all of the inner states. Abstract classes would solve this though
2. I feel like internal states should be implementation specific (pagerState for example), though if you disagree I can abstract a class for this

Reply:
Have created an abstract class